src/resolution_functions/models/pychop.py

```diff
@@ -429,25 +434,24 @@ def get_chopper_width_squared(cls,
         return sigma * SIGMA2FWHMSQ, None
 
 
-class PyChopModelNonFermi(PyChopModel):
+class PyChopModelNonFermi(PyChopModel, ABC):
```

Comment:
`PyChopModelNonFermi` provides a concrete implementation for the abstract method `_get_chopper_width_squared` defined in `PyChopModel`, which ig should work with a mixin, but might be unclear. Theoretically, we could define and abstract __init__ to enforce the behaviour you suggested, something like
```
class PyChopModelNonFermi:
    @abstractmethod
    def __ini__(self, model_data: PyChopModelDataNonFermi, e_init: Optional[float] = None, **_):
        pass
```

I have been putting off writing documentation until the API is stable, but ig we consider it fairly stable now, so it could be a good time to start: the question would be whether it would be better to do it piecewise, starting in this PR, or whether to do it all at once on a separate branch.

Reply:
It's never too early for classes and functions to have docstrings, especially if they are public. I don't mind leaving the user guide / examples for later: but thanks to autodocs, usable documentation can come together very quickly if we write good docstrings along the way.

Reply:
Does that mean you want to start adding docstrings from this PR, rather than doing it all at once in a dedicated PR?

Also, since you mention public, I suppose `PyChopModel` and `PyChopModelNonFermi` should probably be private, right? They are abstract base classes that the user should never see; only `PyChopModelFermi`, `PyChopModelCNCS`, and `PyChopModelLET` are the concrete classes that people will be interacting with.

Reply:
New code should have docstrings. If existing code that is not otherwise touched by this PR needs a docstring, it should be added in a separate PR to simplify review. It's ok if some "trivial" functions don't have complete docstrings.

PyChopModel / PyChopModelNonFermi could be public if it is reasonable for people to write further PyChop instruments on top of them. But by keeping them private we are more free to play with the API in refactoring. I think an argument can be made either way; it is up to you.